tests/test_interpolation_cubic.py

```diff
@@ -0,0 +1,269 @@
+    """Create a set of sparse points from a set of coordinate
+    ranges for each spatial dimension.
+    """
+    points = SparseFunction(name=name, grid=grid, npoint=npoints, cubic=cubic, sinc=sinc)
+    for i, r in enumerate(ranges):
+        points.coordinates.data[:, i] = np.linspace(r[0], r[1], npoints)
+    return points
+
+
+def time_points(grid, ranges, npoints, name='points', nt=10, cubic=None, sinc=None):
+    """Create a set of sparse points from a set of coordinate
+    ranges for each spatial dimension.
+    """
+    points = SparseTimeFunction(name=name, grid=grid, npoint=npoints,
+                                nt=nt, cubic=cubic, sinc=sinc)
+    for i, r in enumerate(ranges):
+        points.coordinates.data[:, i] = np.linspace(r[0], r[1], npoints)
+    return points
+
+
+def custom_points(grid, ranges, npoints, name='points', cubic=None, sinc=None):
+    """Create a set of sparse points from a set of coordinate
+    ranges for each spatial dimension.
+    """
+    scale = Dimension(name="scale")
+    dim = Dimension(name="dim")
+    points = SparseFunction(name=name, grid=grid, dimensions=(scale, dim),
+                            shape=(3, npoints), npoint=npoints, cubic=cubic, sinc=sinc)
+    for i, r in enumerate(ranges):
+        points.coordinates.data[:, i] = np.linspace(r[0], r[1], npoints)
+    return points
+
+
+@pytest.mark.parametrize('shape, coords', [
+    ((101, 101), [(3.4, 95.), (3., 92)]),
+    ((101, 101, 101), [(3.4, 95.), (3., 92), (10., 10.)])
+])
+def test_interpolate(shape, coords, npoints=20):
+    """Test generic point interpolation testing the x-coordinate of an
+    abitrary set of points going across the grid.
+    """
+
+    a = exp_box(shape=shape)
+    p = points(a.grid, coords, npoints=npoints, cubic=True)
+    expr = p.interpolate(a)
+    op = Operator(expr)
+    # print(p.coordinates.data)
+    op(a=a)
+
+    # Precisão 3D nao bate em 10-6 mas sim em 10-5
+    expected_values = [point[1]**2 for point in p.coordinates.data]
+
+    assert np.allclose(p.data[:], expected_values, atol=1e-5)
+
+
+@pytest.mark.parametrize('shape, coords', [
+    ((101, 101), [(3.4, 95.), (3., 92)]),
+    ((101, 101, 101), [(3.4, 95.), (3., 92), (10., 10.)])
+])
+def test_interpolate_cumm(shape, coords, npoints=20):
+    """Test generic point interpolation testing the x-coordinate of an
+    abitrary set of points going across the grid.
+    """
+    a = exp_box(shape=shape)
+    p = points(a.grid, coords, npoints=npoints, cubic=True)
+
+    p.data[:] = 1.
+    expr = p.interpolate(a, increment=True)
+    Operator(expr)(a=a)
+
+    expected_values = [((point[1]**2) + 1) for point in p.coordinates.data]
+
+    assert np.allclose(p.data[:], expected_values, rtol=1e-5)
+
+
+@pytest.mark.parametrize('shape, coords', [
+    ((101, 101), [(3.4, 95.), (3., 92)]),
+    ((101, 101, 101), [(3.4, 95.), (3., 92), (10., 10.)])
+])
+def test_interpolate_time_shift(shape, coords, npoints=20):
```

Comment:
Are these all from `test_interpolation`? This looks like it could be a single `parametrize` change there to test linear and cubic cases.